dashboard/app/controllers/courses_controller.rb

```diff
@@ -8,9 +10,16 @@ def index
   end
 
   def show
-    course_name = params[:course_name].tr('-', '_').titleize
-    course = Course.find_by_name(course_name)
-    raise ActiveRecord::RecordNotFound unless course
+    course = Course.find_by_name(params[:course_name])
+    unless course
+      # PLC courses have different ways of getting to name. ideally this goes
+      # away eventually
+      course_name = params[:course_name].tr('-', '_').titleize
+      course = Course.find_by_name!(course_name)
+      # only support this alternative course name for plc courses
+      raise ActiveRecord::RecordNotFound unless course.plc_course
```

Comment:
The bang method `find_by_name!` on line 18 will raise this already. Either change that to the non-bang version, and explicitly raise here, or remove this (and know it will raise in all cases even for plc_course